src/main/scala/mesosphere/marathon/storage/migration/MigrationTo152.scala

```diff
@@ -0,0 +1,61 @@
+package mesosphere.marathon
+package storage.migration
+
+import akka.Done
+import akka.stream.Materializer
+import akka.stream.scaladsl.{ Flow, Keep, Sink }
+import com.typesafe.scalalogging.StrictLogging
+import mesosphere.marathon.core.instance.Instance
+import mesosphere.marathon.state._
+import mesosphere.marathon.storage.migration.MigrationTo146.Environment
+import mesosphere.marathon.storage.repository.InstanceRepository
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+@SuppressWarnings(Array("ClassNames"))
+class MigrationTo152(instanceRepository: InstanceRepository)(implicit ctx: ExecutionContext, mat: Materializer) extends StrictLogging {
+
+  def migrate(): Future[Done] = {
+    MigrationTo152.migrateUnreachableInstances(instanceRepository)(Environment(sys.env), ctx, mat)
+  }
+}
+
+object MigrationTo152 extends StrictLogging {
+
+  private def changeUnreachableStrategyForInstances(instance: Instance): Instance = {
+    instance.copy(unreachableStrategy = MigrationTo146.changeUnreachableStrategy(instance.unreachableStrategy))
+  }
+
+  def migrateUnreachableInstances(instanceRepository: InstanceRepository)(implicit env: Environment, ctx: ExecutionContext, mat: Materializer): Future[Done] = {
+    logger.info("Starting unreachable strategy migration to 1.5.2")
+
+    val instanceSink =
+      Flow[Instance]
+        .mapAsync(Int.MaxValue)(instanceRepository.store)
```

Comment:
Do you think it makes sense to restrict amount of simultaneous calls to some reasonable default? Let's say 8 for example (must common number of CPUs * 2).

Reply:
+1 `Int.MaxValue` seem to big.

Reply:
👍